packages/blockchain-wallet-v4/src/redux/walletSync/middleware.js

```diff
@@ -1,18 +1,88 @@
 import { futurizeP } from 'futurize'
 import Task from 'data.task'
-import { compose } from 'ramda'
+import { compose, assoc, join, pipe, curry, map, range,
+  identity, converge, add, keysIn } from 'ramda'
+import { networks } from 'bitcoinjs-lib'
 
 import * as A from '../actions'
 import * as T from '../actionTypes'
-import { Wrapper } from '../../types'
+import { Wrapper, Wallet, HDAccount } from '../../types'
 import * as selectors from '../selectors'
 
+/**
+ * Number of unused addresses for each HD Account
+ * to sync with platform
+ */
+export const unusedAddressesCount = 10
+
+/**
+ * Launches derivation of future addresses for target HDAccount
+ * @curried
+ * @param {any} state redux state
+ * @param {HDAccount.HDAccount} account
+ * @returns {Promise<String>[]} array of promises, each of them resolves into address
+ */
```

Comment:
I never get used to that kind of annotation. I think they are super noisy. 
I tend to use the type annotation:
// getHDAccountAddressPromises :: state -> Account -> [Promise(Address)]